src/System.Collections/src/System/Collections/Generic/Queue.cs

```diff
@@ -181,17 +181,36 @@ void ICollection.CopyTo(Array array, int index)
         }
 
         // Adds item to the tail of the queue.
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
Same comment as on `Stack<T>.Push` applies here.
For `List<T>.Add` aggressive inlining makes sense, as it is just a comparison for hot/cold-path, then on the hot-path an array access plus incrementing the version.

Here more work needs to be done, so the call-site gets quite a bit bigger.

Reply:
I've removed `AggressiveInlining` as per the discussion above.